crates/prover/src/runner.rs

```diff
@@ -411,9 +408,27 @@ where
                 .unwrap();
 
             let mut sequencer_commitments = Vec::<SequencerCommitment>::new();
-            let mut zk_proofs = Vec::<Proof>::new();
 
-            self.extract_relevant_l1_data(l1_block, &mut sequencer_commitments, &mut zk_proofs);
+            let mut da_data = self.da_service.extract_relevant_blobs(l1_block);
+            // if we don't do this, the zk circuit can't read the sequencer commitments
+            da_data.iter_mut().for_each(|blob| {
+                blob.full_data();
+            });
+            da_data.iter_mut().for_each(|tx| {
```

Comment:
There is no need to extract `zk_proofs` in prover as it does not process them. 

This code has been simplfied